Add File purposes as constants

revett opened the pull request:
## Describe the change

- When uploading a `File`, you need to define a `.Purpose`
- Currently this is a `string` that is passed to the `FileRequest` type
- It would be nicer to have these as constants

<img width="529" alt="image" src="https://github.com/sashabaranov/go-openai/assets/2796074/135711b0-9dab-4131-94dc-e61c7164e401">

> See: https://platform.openai.com/docs/api-reference/files

## Describe your solution

Add a constant for each supported purpose as defined in the OpenAI API.

## Tests

No tests needed.

## Additional context

Not related to an issue, just something I saw when using the new Assistants API.

floodfx commented:
I added something similar in #568 but with more type safety.  We should probably merge these efforts?  I can add the additional types to #568.  I am unclear whether the type safe approach I used is typical of this project though?

revett commented:
> I added something similar in #568 but with more type safety. We should probably merge these efforts? I can add the additional types to #568. I am unclear whether the type safe approach I used is typical of this project though?

Thanks! I've rebased your merged changes in to this PR. Looks like we still need two more purposes added.

@sashabaranov This is ready for review 👍

sashabaranov reviewed (approved):
Thank you!
